care/facility/api/serializers/patient_consultation.py

```diff
@@ -196,6 +201,88 @@ def validate_bed_number(self, bed_number):
+                "bp": {
+                    "type": "object",
+                    "properties": {
+                        "property1": {"type": "null"},
+                        "property2": {"type": "null"},
+                    },
+                    "nullable": True,
+                },
+                "pulse": {"type": "number", "nullable": True, "default": 72},
+                "temperature": {"type": "string", "nullable": True},
+                "consciousness_level": {
+                    "type": "string",
+                    "nullable": True,
+                    "default": 0,
+                },
+                "modified_date": {
+                    "type": "string",
+                    "format": "date-time",
+                    "nullable": True,
+                },
+            },
+        }
+    )
+    def get_mews_field(self, consultation):
+        current_time = localtime(now())
+        past_30_minutes = current_time - timedelta(minutes=30)
+        mews_field_data = {
+            "resp": None,
+            "bp": {},
+            "pulse": None,
+            "temperature": None,
+            "consciousness_level": DailyRound.ConsciousnessType.UNKNOWN.name,
+            "modified_date": None,
+        }
+        recentDailyArray = (
+            DailyRound.objects.filter(
+                consultation=consultation,
+                created_date__gte=past_30_minutes,
+                created_date__lte=current_time,
+            )
+            .order_by("-created_date")
+            .only(*(list(mews_field_data.keys())))
+        )
+        if len(recentDailyArray) == 0:
+            return mews_field_data
+        mews_field_data["modified_date"] = localtime(recentDailyArray[0].modified_date)
+        for obj in recentDailyArray:
+            count = 0
+            for key in mews_field_data:
+                newValue = getattr(obj, key, None)
+                if (
+                    (key != "modified_date")
+                    and (
+                        key == "bp"
+                        and len(mews_field_data[key]) == 0
+                        and len(newValue) == 3
+                    )
+                    or (
+                        key == "consciousness_level"
+                        and mews_field_data[key]
+                        == DailyRound.ConsciousnessType.UNKNOWN.name
+                        and newValue != DailyRound.ConsciousnessType.UNKNOWN.value
+                    )
+                    or (mews_field_data[key] is None and newValue is not None)
+                ):
+                    mews_field_data[key] = (
+                        None
+                        if newValue == None
+                        else DailyRound.ConsciousnessType(newValue).name
+                        if key == "consciousness_level"
+                        else newValue
+                    )
+                    count += 1
+            if count == 5:
+                break
```

Comment:
Code seems to be not very readable / more complicated than necessary... We simply need to find the latest non-null value for specific fields from the filtered queryset right.

You could also maybe simply use subquery and coalesce.

Refer: 

https://docs.djangoproject.com/en/5.0/ref/models/expressions/#subquery-expressions
https://docs.djangoproject.com/en/5.0/ref/models/database-functions/#django.db.models.functions.Coalesce
https://docs.djangoproject.com/en/5.0/ref/models/querysets/#django.db.models.query.QuerySet.annotate

Reply:
I made changes regarding its readibility, please verify